### ISSUE TITLE
a-shell keeps forgetting custom bookmarked locations

### ISSUE THREAD

**Issue by NightMarekApps** [holzschu/a-shell#460]
hi! since the last update the app has forgotten my bookmarked locations multiple times. i dont know what causes it

**Comment by holzschu** [comment]
What I've done in the last update is check that bookmarks are still valid at startup, and remove the ones that aren't. This was to avoid the "cd ~bookmark" that was followed by "no such file or directory". It shouldn't remove a bookmark that is still valid.

**Comment by NightMarekApps** [comment]
i did some testing to see when that happens. it happens everytime i restart my device. bookmarks disappear and i get this jump: <bookmark> not found

**Comment by NightMarekApps** [comment]
![IMG_0103](https://user-images.githubusercontent.com/84610465/188804739-713ed55b-4345-4c7a-803f-a5919d7b69e8.png)
i restarted the device before using cd command above
happens both on ipad and iphone (both beta) since the latest a-shell update

**Comment by holzschu** [comment]
Yes, restarting the device was making the bookmarks invalid, so that "cd ~bookmarkName" was returning "No such file or directory". You're saying that it was working with jump before, even though it was not working with cd?

**Comment by NightMarekApps** [comment]
i only ever used jump and it always worked for me. since last update this happens after restart

**Comment by holzschu** [comment]
So the problem was with cd, not with the bookmarks. Thank you very much. I'm going to change that.

**Comment by NightMarekApps** [comment]
Thank you. i dont quite understand how cd is the problem here
my workflow is this:
pick folder, bookmark it once with custom shortname using bookmark command. then always go to that folder using the jump command. now after restart it says jump: shortname not found

im happy to do some further testing if you need

**Comment by holzschu** [comment]
`cd` was checking both `isReadableFileAtPath` and `changeCurrentDirectoryPath` for each directory, jump was just trying `changeCurrentDirectoryPath`. It seems that after a reboot, `isReadableFileAtPath` becomes false for a bookmarked directory, even though `changeCurrentDirectoryPath` still works. That's weird and unexpected, but I'll adapt the code.

**Comment by NightMarekApps** [comment]
Thank you. I did some more testing though and after bookmarking locations, i ran showmarks to show my newly created bookmarks, and they are listed there

i then rebooted and ran showmarks again, and these bookmarks are gone. they disappear after restart

**Comment by holzschu** [comment]
That makes sense: that's what I wrote the code for, to avoid the issue with "cd ~bookmark: permission denied".

**Comment by NightMarekApps** [comment]
im sorry that is complicated for me, does that mean your fix will fix it? or that it forgets them by design :)

**Comment by holzschu** [comment]
Yes, the new fix will remove the previous fix, and you will get back the previous behaviour with version 1.9.7. As a bonus, people who use "cd ~bookmark" instead of "jump bookmark" will also enjoy bookmarks that remain valid after a reboot.

**Comment by NightMarekApps** [comment]
Thank you so much for all this and for quick responses!

**Comment by CodingAlbert** [comment]
When can we expect a new release with the fix?

**Comment by s3fs** [comment]
Still happens for me @ v1.16.1. Deleting all bookmarks doesn’t help (`deletemark —all` -> `pickFolder` -> `bookmark` -> `already bookmarked`; `not found` when trying to `jump`. `cd` behaves in the same manner.
